optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -275,12 +278,19 @@ def gaudi_llama_repeat_kv(
 
 #  FusedScaledDotProductAttention
 class ModuleFusedSDPA(torch.nn.Module):
-    def __init__(self, fusedSDPA):
+    def __init__(self, fusedSDPA, scale, attention_dropout, enable_recompute=False):
         super().__init__()
         self._hpu_kernel_fsdpa = fusedSDPA
-
-    def forward(self, query, key, value, attn_mask, dropout_p, is_casual, scale, softmax_mode):
-        return self._hpu_kernel_fsdpa.apply(query, key, value, attn_mask, dropout_p, is_casual, scale, softmax_mode)
+        self.scale = scale
```

Comment:
Why are we storing scale, attention_dropout and enable_recompute in class object? These are already being passed to forward method anyway.

Reply:
This is needed for the module from TE